cmd/wallet-web/src/pages/VaultsPage.vue

```diff
@@ -3,6 +3,140 @@
+const showDeleteModal = ref(false);
+const showRenameModal = ref(false);
+const selectedVaultId = ref('');
+const authToken = ref(null);
+const credentialManager = ref(null);
+const collectionManager = ref(null);
+const walletUser = ref(null);
+const skipWelcomeMsg = computed(() => store?.state?.user?.preference?.skipWelcomeMsg);
+const existingNames = computed(() => vaults.value.map((vault) => vault.name));
+
+// Store Getters
+const getCurrentUser = computed(() => store.getters['getCurrentUser']);
+const getAgentInstance = computed(() => store.getters['agent/getInstance']);
+
+// Store Actions
+const refreshUserPreference = () => store.dispatch('refreshUserPreference');
+
+// Methods
+onMounted(async () => {
+  const { profile } = getCurrentUser.value;
+  const { user, token } = profile;
+  authToken.value = token;
+  credentialManager.value = new CredentialManager({ agent: getAgentInstance.value, user });
+  collectionManager.value = new CollectionManager({ agent: getAgentInstance.value, user });
+  walletUser.value = new WalletUser({ agent: getAgentInstance.value, user });
+  // Fetch all the credentials stored.
+  // TODO: Issue-1250 Refactor to not to save credentials without vault ID.
+  await getNumOfCreds();
+  await fetchVaults();
+  loading.value = false;
+  watchEffect(async () => {
+    if (vaultsStore.vaultsOutdated) {
+      loading.value = true;
+      await fetchVaults();
+      vaultsMutations.setVaultsOutdated(false);
+      loading.value = false;
+    }
+  });
+});
+
+function toggleDeleteModal(vaultId) {
+  selectedVaultId.value = vaultId;
+  showDeleteModal.value = !showDeleteModal.value;
+}
+
+function toggleRenameModal(vaultId) {
+  selectedVaultId.value = vaultId;
+  showRenameModal.value = !showRenameModal.value;
+}
+
+async function getNumOfCreds() {
+  // Fetching all credentials
+  const credentials = await credentialManager.value.getAllCredentialMetadata(authToken.value);
+  numOfCreds.value = credentials.length;
+}
+
+async function fetchVaults() {
+  vaults.value = [];
+  // Fetching all vaults
+  const { contents: rawVaults } = await collectionManager.value.getAll(authToken.value);
+
+  const vaultsArray = Object.values(rawVaults);
+  // For each vault get a number of credentials
+  await Promise.all(
+    vaultsArray.map(async (vault) => {
+      // Fetching all credentials stored inside each vault
+      // TODO: #1236 Revisit the solution to avoid getting all the credentials
+      await credentialManager.value
+        .getAllCredentialMetadata(authToken.value, { collection: vault.id })
+        .then((credentials) => {
+          vault['numOfCreds'] = credentials.length;
+          vaults.value.push(vault);
+        });
+    })
+  );
+}
+
+async function updateUserPreferences() {
+  try {
+    // Used to close the welcome message banner instantly in the UI
+    skippedLocally.value = true;
+    await walletUser.value.updatePreferences(authToken.value, { skipWelcomeMsg: true });
+    refreshUserPreference();
```

Comment:
We should await here since it's async
```suggestion
    await refreshUserPreference();
```